Subject: [PATCH] [quarkus_patch_javax_string] Replacing javax -> jakarta
---
Index: drools-quarkus-extension/drools-quarkus-util-deployment/src/main/java/org/drools/drl/quarkus/util/deployment/DroolsQuarkusResourceUtils.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/drools-quarkus-extension/drools-quarkus-util-deployment/src/main/java/org/drools/drl/quarkus/util/deployment/DroolsQuarkusResourceUtils.java b/drools-quarkus-extension/drools-quarkus-util-deployment/src/main/java/org/drools/drl/quarkus/util/deployment/DroolsQuarkusResourceUtils.java
--- a/drools-quarkus-extension/drools-quarkus-util-deployment/src/main/java/org/drools/drl/quarkus/util/deployment/DroolsQuarkusResourceUtils.java	(revision 4eb6d59aefa1df16a286666f75ead10d8b77746f)
+++ b/drools-quarkus-extension/drools-quarkus-util-deployment/src/main/java/org/drools/drl/quarkus/util/deployment/DroolsQuarkusResourceUtils.java	(date 1677837541284)
@@ -222,8 +222,8 @@
     }

     private static final String RULE_UNIT_DEF_PRODUCER =
-            "import javax.enterprise.context.Dependent;\n" +
-                    "import javax.enterprise.inject.Produces;\n" +
+            "import jakarta.enterprise.context.Dependent;\n" +
+                    "import jakarta.enterprise.inject.Produces;\n" +
                     "\n" +
                     "import org.drools.ruleunits.api.RuleUnit;\n" +
                     "import org.drools.ruleunits.api.RuleUnitProvider;\n" +
