diff --git a/data-index/data-index-storage/data-index-storage-oracle/src/main/java/org/kie/kogito/index/oracle/storage/OracleStorageService.java b/data-index/data-index-storage/data-index-storage-oracle/src/main/java/org/kie/kogito/index/oracle/storage/OracleStorageService.java
index 8c1d6dcfa..efadb2929 100644
--- a/data-index/data-index-storage/data-index-storage-oracle/src/main/java/org/kie/kogito/index/oracle/storage/OracleStorageService.java
+++ b/data-index/data-index-storage/data-index-storage-oracle/src/main/java/org/kie/kogito/index/oracle/storage/OracleStorageService.java
@@ -18,7 +18,9 @@
  */
 package org.kie.kogito.index.oracle.storage;

+import jakarta.annotation.Priority;
 import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Alternative;
 import jakarta.inject.Inject;

 import org.kie.kogito.index.model.Job;
@@ -28,14 +30,14 @@ import org.kie.kogito.index.model.UserTaskInstance;
 import org.kie.kogito.persistence.api.Storage;
 import org.kie.kogito.persistence.api.StorageService;

-import io.quarkus.arc.AlternativePriority;
 import io.quarkus.arc.properties.IfBuildProperty;

 import static java.lang.String.format;
 import static org.kie.kogito.persistence.api.factory.Constants.PERSISTENCE_TYPE_PROPERTY;
 import static org.kie.kogito.persistence.oracle.Constants.ORACLE_STORAGE;

-@AlternativePriority(1)
+@Alternative
+@Priority(1)
 @ApplicationScoped
 @IfBuildProperty(name = PERSISTENCE_TYPE_PROPERTY, stringValue = ORACLE_STORAGE)
 public class OracleStorageService implements StorageService {
diff --git a/data-index/data-index-storage/data-index-storage-postgresql/src/main/java/org/kie/kogito/index/postgresql/storage/PostgreSqlStorageService.java b/data-index/data-index-storage/data-index-storage-postgresql/src/main/java/org/kie/kogito/index/postgresql/storage/PostgreSqlStorageService.java
index deab75186..93bb36faf 100644
--- a/data-index/data-index-storage/data-index-storage-postgresql/src/main/java/org/kie/kogito/index/postgresql/storage/PostgreSqlStorageService.java
+++ b/data-index/data-index-storage/data-index-storage-postgresql/src/main/java/org/kie/kogito/index/postgresql/storage/PostgreSqlStorageService.java
@@ -18,7 +18,9 @@
  */
 package org.kie.kogito.index.postgresql.storage;

+import jakarta.annotation.Priority;
 import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Alternative;
 import jakarta.inject.Inject;

 import org.kie.kogito.index.model.Job;
@@ -28,14 +30,14 @@ import org.kie.kogito.index.model.UserTaskInstance;
 import org.kie.kogito.persistence.api.Storage;
 import org.kie.kogito.persistence.api.StorageService;

-import io.quarkus.arc.AlternativePriority;
 import io.quarkus.arc.properties.IfBuildProperty;

 import static java.lang.String.format;
 import static org.kie.kogito.persistence.api.factory.Constants.PERSISTENCE_TYPE_PROPERTY;
 import static org.kie.kogito.persistence.postgresql.Constants.POSTGRESQL_STORAGE;

-@AlternativePriority(1)
+@Alternative
+@Priority(1)
 @ApplicationScoped
 @IfBuildProperty(name = PERSISTENCE_TYPE_PROPERTY, stringValue = POSTGRESQL_STORAGE)
 public class PostgreSqlStorageService implements StorageService {
diff --git a/trusty/trusty-storage/trusty-storage-postgresql/src/main/java/org/kie/kogito/trusty/storage/postgresql/PostgresStorageService.java b/trusty/trusty-storage/trusty-storage-postgresql/src/main/java/org/kie/kogito/trusty/storage/postgresql/PostgresStorageService.java
index 6e174a82a..2adb6a2db 100644
--- a/trusty/trusty-storage/trusty-storage-postgresql/src/main/java/org/kie/kogito/trusty/storage/postgresql/PostgresStorageService.java
+++ b/trusty/trusty-storage/trusty-storage-postgresql/src/main/java/org/kie/kogito/trusty/storage/postgresql/PostgresStorageService.java
@@ -18,7 +18,9 @@
  */
 package org.kie.kogito.trusty.storage.postgresql;

+import jakarta.annotation.Priority;
 import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Alternative;
 import jakarta.inject.Inject;

 import org.kie.kogito.explainability.api.CounterfactualExplainabilityRequest;
@@ -29,15 +31,15 @@ import org.kie.kogito.persistence.api.StorageService;
 import org.kie.kogito.trusty.storage.api.model.decision.DMNModelWithMetadata;
 import org.kie.kogito.trusty.storage.api.model.decision.Decision;

-import io.quarkus.arc.AlternativePriority;
 import io.quarkus.arc.properties.IfBuildProperty;

 import static java.lang.String.format;
 import static org.kie.kogito.persistence.api.factory.Constants.PERSISTENCE_TYPE_PROPERTY;
 import static org.kie.kogito.persistence.postgresql.Constants.POSTGRESQL_STORAGE;

+@Alternative
+@Priority(1)
 @ApplicationScoped
-@AlternativePriority(1)
 @IfBuildProperty(name = PERSISTENCE_TYPE_PROPERTY, stringValue = POSTGRESQL_STORAGE)
 public class PostgresStorageService implements StorageService {

