commit 8e39c23e777e3e6cc8124bc4d5f07f871ce8f38b
Author: radtriste <radtriste@jenkins.redhat>
Date:   Wed Apr 19 09:44:15 2023 +0200

    patch: Quarkus Kubernetes update

diff --git a/quarkus/addons/knative/eventing/deployment/src/test/java/org/kie/kogito/addons/quarkus/knative/eventing/deployment/KogitoProcessKnativeEventingProcessorTest.java b/quarkus/addons/knative/eventing/deployment/src/test/java/org/kie/kogito/addons/quarkus/knative/eventing/deployment/KogitoProcessKnativeEventingProcessorTest.java
index cafd543a0c..a65d976efe 100644
--- a/quarkus/addons/knative/eventing/deployment/src/test/java/org/kie/kogito/addons/quarkus/knative/eventing/deployment/KogitoProcessKnativeEventingProcessorTest.java
+++ b/quarkus/addons/knative/eventing/deployment/src/test/java/org/kie/kogito/addons/quarkus/knative/eventing/deployment/KogitoProcessKnativeEventingProcessorTest.java
@@ -33,6 +33,7 @@ import org.kie.kogito.event.cloudevents.CloudEventMeta;
 import org.kie.kogito.quarkus.extensions.spi.deployment.KogitoProcessContainerGeneratorBuildItem;
 
 import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.kubernetes.spi.DeployStrategy;
 import io.quarkus.kubernetes.spi.KubernetesDeploymentTargetBuildItem;
 import io.quarkus.kubernetes.spi.KubernetesResourceMetadataBuildItem;
 
@@ -105,7 +106,7 @@ class KogitoProcessKnativeEventingProcessorTest {
         final MockKogitoKnativeMetadataProducer metadata = new MockKogitoKnativeMetadataProducer();
         final KubernetesResourceMetadataBuildItem kubernetesResourceMetadataBuildItem = new KubernetesResourceMetadataBuildItem("kubernetes", "apps", "v1", "Deployment", "name");
         final List<KubernetesResourceMetadataBuildItem> kubernetesMetaBuildItems = singletonList(kubernetesResourceMetadataBuildItem);
-        final List<KubernetesDeploymentTargetBuildItem> deploymentTargets = singletonList(new KubernetesDeploymentTargetBuildItem("kubernetes", "Deployment", "apps", "v1"));
+        final List<KubernetesDeploymentTargetBuildItem> deploymentTargets = singletonList(new KubernetesDeploymentTargetBuildItem("kubernetes", "Deployment", "apps", "v1", DeployStrategy.CreateOrUpdate));
         final List<KogitoCloudEventsBuildItem> extendedCloudEventsBuildItems = singletonList(new KogitoCloudEventsBuildItem(Set.of(EXTENDED_CLOUD_EVENT1, EXTENDED_CLOUD_EVENT2)));
 
         doCallRealMethod().when(processor).buildMetadata(singletonList(containerGeneratorBuildItem), extendedCloudEventsBuildItems, deploymentTargets, kubernetesMetaBuildItems, metadata);
