diff --git a/trusty/trusty-service/trusty-service-common/pom.xml b/trusty/trusty-service/trusty-service-common/pom.xml
index 326b62576..33d2933e4 100644
--- a/trusty/trusty-service/trusty-service-common/pom.xml
+++ b/trusty/trusty-service/trusty-service-common/pom.xml
@@ -65,6 +65,10 @@
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-oidc</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.smallrye.reactive</groupId>
+      <artifactId>mutiny-zero-flow-adapters</artifactId>
+    </dependency>
 
     <dependency>
       <groupId>io.quarkus</groupId>
diff --git a/trusty/trusty-service/trusty-service-common/src/main/java/org/kie/kogito/trusty/service/common/messaging/outgoing/ExplainabilityRequestProducer.java b/trusty/trusty-service/trusty-service-common/src/main/java/org/kie/kogito/trusty/service/common/messaging/outgoing/ExplainabilityRequestProducer.java
index b031c22bd..eeba524c8 100644
--- a/trusty/trusty-service/trusty-service-common/src/main/java/org/kie/kogito/trusty/service/common/messaging/outgoing/ExplainabilityRequestProducer.java
+++ b/trusty/trusty-service/trusty-service-common/src/main/java/org/kie/kogito/trusty/service/common/messaging/outgoing/ExplainabilityRequestProducer.java
@@ -20,6 +20,7 @@ import java.net.URI;
 import java.util.Optional;
 
 import jakarta.enterprise.context.ApplicationScoped;
+import mutiny.zero.flow.adapters.AdaptersToReactiveStreams;
 
 import org.eclipse.microprofile.reactive.messaging.Outgoing;
 import org.kie.kogito.event.cloudevents.utils.CloudEventUtils;
@@ -53,6 +54,6 @@ public class ExplainabilityRequestProducer {
 
     @Outgoing("trusty-explainability-request")
     public Publisher<String> getEventPublisher() {
-        return eventSubject.toHotStream();
+        return AdaptersToReactiveStreams.publisher(eventSubject.toHotStream());
     }
 }
diff --git a/trusty/trusty-service/trusty-service-common/src/test/java/org/kie/kogito/trusty/service/common/messaging/outgoing/ExplainabilityRequestProducerTest.java b/trusty/trusty-service/trusty-service-common/src/test/java/org/kie/kogito/trusty/service/common/messaging/outgoing/ExplainabilityRequestProducerTest.java
index 39af2fb0b..a2825c050 100644
--- a/trusty/trusty-service/trusty-service-common/src/test/java/org/kie/kogito/trusty/service/common/messaging/outgoing/ExplainabilityRequestProducerTest.java
+++ b/trusty/trusty-service/trusty-service-common/src/test/java/org/kie/kogito/trusty/service/common/messaging/outgoing/ExplainabilityRequestProducerTest.java
@@ -23,6 +23,7 @@ import org.kie.kogito.explainability.api.LIMEExplainabilityRequest;
 import org.kie.kogito.explainability.api.ModelIdentifier;
 
 import io.smallrye.mutiny.helpers.test.AssertSubscriber;
+import mutiny.zero.flow.adapters.AdaptersToReactiveStreams;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
@@ -33,7 +34,7 @@ class ExplainabilityRequestProducerTest {
         AssertSubscriber<String> subscriber = AssertSubscriber.create(1);
 
         ExplainabilityRequestProducer producer = new ExplainabilityRequestProducer();
-        producer.getEventPublisher().subscribe(subscriber);
+        producer.getEventPublisher().subscribe(AdaptersToReactiveStreams.subscriber(subscriber));
 
         producer.sendEvent(new LIMEExplainabilityRequest(
                 "executionId", "http://localhost:8080/model",
