commit 5a4c65082e96780939a5f3011b46c048c4fbaa8d
Author: radtriste <radtriste@jenkins.redhat>
Date:   Wed Apr 19 10:06:18 2023 +0200

    patch: reactivestreams

diff --git a/quarkus/addons/tracing-decision/runtime/src/main/java/org/kie/kogito/tracing/decision/QuarkusModelEventEmitter.java b/quarkus/addons/tracing-decision/runtime/src/main/java/org/kie/kogito/tracing/decision/QuarkusModelEventEmitter.java
index 0868563be8..0934b9ec3a 100644
--- a/quarkus/addons/tracing-decision/runtime/src/main/java/org/kie/kogito/tracing/decision/QuarkusModelEventEmitter.java
+++ b/quarkus/addons/tracing-decision/runtime/src/main/java/org/kie/kogito/tracing/decision/QuarkusModelEventEmitter.java
@@ -18,9 +18,10 @@ package org.kie.kogito.tracing.decision;
 import jakarta.inject.Inject;
 import jakarta.inject.Singleton;
 
+import java.util.concurrent.Flow.Publisher;
+
 import org.eclipse.microprofile.reactive.messaging.Outgoing;
 import org.kie.kogito.decision.DecisionModelResourcesProvider;
-import org.reactivestreams.Publisher;
 
 import io.smallrye.mutiny.operators.multi.processors.BroadcastProcessor;
 
diff --git a/quarkus/addons/tracing-decision/runtime/src/main/java/org/kie/kogito/tracing/decision/QuarkusTraceEventEmitter.java b/quarkus/addons/tracing-decision/runtime/src/main/java/org/kie/kogito/tracing/decision/QuarkusTraceEventEmitter.java
index a887b5519b..1558660d9b 100644
--- a/quarkus/addons/tracing-decision/runtime/src/main/java/org/kie/kogito/tracing/decision/QuarkusTraceEventEmitter.java
+++ b/quarkus/addons/tracing-decision/runtime/src/main/java/org/kie/kogito/tracing/decision/QuarkusTraceEventEmitter.java
@@ -17,9 +17,10 @@ package org.kie.kogito.tracing.decision;
 
 import jakarta.inject.Singleton;
 
+import java.util.concurrent.Flow.Publisher;
+
 import org.eclipse.microprofile.reactive.messaging.Outgoing;
 import org.kie.kogito.tracing.EventEmitter;
-import org.reactivestreams.Publisher;
 
 import io.smallrye.mutiny.operators.multi.processors.BroadcastProcessor;
 
