diff --git a/kie-test-util/src/main/java/org/kie/test/util/db/internal/PoolingDataSourceFactory.java b/kie-test-util/src/main/java/org/kie/test/util/db/internal/PoolingDataSourceFactory.java
index 26d202d2f7..7bb32d6311 100644
--- a/kie-test-util/src/main/java/org/kie/test/util/db/internal/PoolingDataSourceFactory.java
+++ b/kie-test-util/src/main/java/org/kie/test/util/db/internal/PoolingDataSourceFactory.java
@@ -28,7 +28,7 @@ import javax.sql.XAConnection;
 import javax.sql.XADataSource;
 import jakarta.transaction.TransactionManager;
 import jakarta.transaction.TransactionSynchronizationRegistry;
-import jakarta.transaction.xa.XAResource;
+import javax.transaction.xa.XAResource;
 
 import com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule;
 import com.arjuna.ats.jta.recovery.XAResourceRecoveryHelper;
