commit a06a83c2f61d7a0ea6a17ac0f7655a6c0e939b82
Author: radtriste <radtriste@jenkins.redhat>
Date:   Wed Apr 19 10:32:26 2023 +0200

    patch: javax.servlet -> jakarta.servlet

diff --git a/kogito-codegen-modules/kogito-codegen-sample/kogito-codegen-sample-generator/src/main/resources/class-templates/SampleRestResourceSpringTemplate.java b/kogito-codegen-modules/kogito-codegen-sample/kogito-codegen-sample-generator/src/main/resources/class-templates/SampleRestResourceSpringTemplate.java
index e1afd2a762..b2bda10310 100644
--- a/kogito-codegen-modules/kogito-codegen-sample/kogito-codegen-sample-generator/src/main/resources/class-templates/SampleRestResourceSpringTemplate.java
+++ b/kogito-codegen-modules/kogito-codegen-sample/kogito-codegen-sample-generator/src/main/resources/class-templates/SampleRestResourceSpringTemplate.java
@@ -15,7 +15,7 @@
  */
 package org.kie.dmn.kogito.quarkus.example;
 
-import javax.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpServletResponse;
 
 import org.kie.kogito.Application;
 import org.kie.kogito.codegen.sample.core.SampleRuntime;
