diff --git a/jbpm/jbpm-bpmn2/src/test/resources/BPMN2-ServiceProcessWithMvelJaxbTransformation.bpmn2 b/jbpm/jbpm-bpmn2/src/test/resources/BPMN2-ServiceProcessWithMvelJaxbTransformation.bpmn2
index fd4c6b32ed..f4d4503a87 100755
--- a/jbpm/jbpm-bpmn2/src/test/resources/BPMN2-ServiceProcessWithMvelJaxbTransformation.bpmn2
+++ b/jbpm/jbpm-bpmn2/src/test/resources/BPMN2-ServiceProcessWithMvelJaxbTransformation.bpmn2
@@ -23,7 +23,7 @@
 
   <process processType="Private" isExecutable="true" id="ServiceProcess" name="Service Process" >
     <extensionElements>
-      <tns:import name="javax.xml.bind.JAXBContext"/>
+      <tns:import name="jakarta.xml.bind.JAXBContext"/>
       <tns:import name="java.io.StringReader"/>
       <tns:import name="java.io.StringWriter"/>
       <tns:import name="org.jbpm.bpmn2.objects.Person"/>
