diff --git a/springboot/archetype/src/main/resources/archetype-resources/src/test/java/GreetingsTest.java b/springboot/archetype/src/main/resources/archetype-resources/src/test/java/GreetingsTest.java
index 100dd1342e..f5e7218e1b 100644
--- a/springboot/archetype/src/main/resources/archetype-resources/src/test/java/GreetingsTest.java
+++ b/springboot/archetype/src/main/resources/archetype-resources/src/test/java/GreetingsTest.java
@@ -20,7 +20,7 @@ import io.restassured.http.ContentType;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 import org.springframework.test.context.junit4.SpringRunner;
diff --git a/springboot/archetype/src/main/resources/archetype-resources/src/test/java/TrafficViolationTest.java b/springboot/archetype/src/main/resources/archetype-resources/src/test/java/TrafficViolationTest.java
index 78097f1ba0..ad5c428f94 100644
--- a/springboot/archetype/src/main/resources/archetype-resources/src/test/java/TrafficViolationTest.java
+++ b/springboot/archetype/src/main/resources/archetype-resources/src/test/java/TrafficViolationTest.java
@@ -23,7 +23,7 @@ import io.restassured.http.ContentType;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 import org.springframework.test.context.junit4.SpringRunner;
diff --git a/springboot/archetype/src/main/resources/archetype-resources/pom.xml b/springboot/archetype/src/main/resources/archetype-resources/pom.xml
index 3fc02e2ac6..86664a7b41 100644
--- a/springboot/archetype/src/main/resources/archetype-resources/pom.xml
+++ b/springboot/archetype/src/main/resources/archetype-resources/pom.xml
@@ -40,6 +40,11 @@
       <artifactId>spring-boot-starter</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-webflux</artifactId>
+    </dependency>
+
     <dependency>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-actuator</artifactId>

