commit 9484c20d8bdd78f9e1be0bdff0d4dcffd1fc6a9a
Author: Pere Fernandez <pefernan@users.noreply.github.com>
Date:   Mon May 15 21:22:56 2023 +0200

    Remove Keycloak Spring Boot Adapter

diff --git a/springboot/bom/pom.xml b/springboot/bom/pom.xml
index 8ee023830e..b495cd9857 100644
--- a/springboot/bom/pom.xml
+++ b/springboot/bom/pom.xml
@@ -25,6 +25,7 @@
     <version.org.springframework.cloud.kubernetes.fabric8>3.0.2</version.org.springframework.cloud.kubernetes.fabric8>
     <!-- CVE-2022-25857: Upgrading snakeyaml until we upgrade to Spring Boot 3.x -->
     <!-- See: https://github.com/spring-projects/spring-boot/issues/32221 -->
+    <version.org.springframework.security>1.0.2</version.org.springframework.security>
   </properties>
   
   <dependencyManagement>
@@ -62,6 +63,11 @@
         <artifactId>spring-cloud-starter-kubernetes-fabric8</artifactId>
         <version>${version.org.springframework.cloud.kubernetes.fabric8}</version>
       </dependency>
+      <dependency>
+        <groupId>org.springframework.security</groupId>
+        <artifactId>spring-security-oauth2-authorization-server</artifactId>
+        <version>${version.org.springframework.security}</version>
+      </dependency>
     </dependencies>
   </dependencyManagement>
   <build>
diff --git a/springboot/test/src/main/java/org/kie/kogito/testcontainers/springboot/KeycloakSpringBootTestResource.java b/springboot/test/src/main/java/org/kie/kogito/testcontainers/springboot/KeycloakSpringBootTestResource.java
index b55ce76dc7..6555a646e7 100644
--- a/springboot/test/src/main/java/org/kie/kogito/testcontainers/springboot/KeycloakSpringBootTestResource.java
+++ b/springboot/test/src/main/java/org/kie/kogito/testcontainers/springboot/KeycloakSpringBootTestResource.java
@@ -15,6 +15,7 @@
  */
 package org.kie.kogito.testcontainers.springboot;
 
+import java.util.HashMap;
 import java.util.Map;
 
 import org.kie.kogito.test.resources.ConditionalSpringBootTestResource;
@@ -29,6 +30,7 @@ import static java.util.Collections.singletonMap;
 public class KeycloakSpringBootTestResource extends ConditionalSpringBootTestResource<KogitoKeycloakContainer> {
 
     public static final String KOGITO_KEYCLOAK_PROPERTY = "keycloak.auth-server-url";
+    public static final String OAUTH2_JWT_ISSUER = "spring.security.oauth2.resourceserver.jwt.issuer-uri";
 
     public KeycloakSpringBootTestResource() {
         super(new KogitoKeycloakContainer());
@@ -36,7 +38,10 @@ public class KeycloakSpringBootTestResource extends ConditionalSpringBootTestRes
 
     @Override
     protected Map<String, String> getProperties() {
-        return singletonMap(KOGITO_KEYCLOAK_PROPERTY, String.format("http://localhost:%s", getTestResource().getMappedPort()));
+        Map<String, String> properties = new HashMap<>();
+        properties.put(KOGITO_KEYCLOAK_PROPERTY, String.format("http://localhost:%s", getTestResource().getMappedPort()));
+        properties.put(OAUTH2_JWT_ISSUER, String.format("http://localhost:%s/realms/%s", getTestResource().getMappedPort(), KogitoKeycloakContainer.REALM));
+        return properties;
     }
 
     public static class Conditional extends KeycloakSpringBootTestResource {
