diff --git a/springboot/addons/messaging/integration-tests/src/it/cloudevents-spring-boot-addon-it-invoked/src/test/java/org/kie/kogito/integrationtests/SpringBootTopicsInformationResourceTest.java b/springboot/addons/messaging/integration-tests/src/it/cloudevents-spring-boot-addon-it-invoked/src/test/java/org/kie/kogito/integrationtests/SpringBootTopicsInformationResourceTest.java
index b30ab871d7..21cd942f16 100644
--- a/springboot/addons/messaging/integration-tests/src/it/cloudevents-spring-boot-addon-it-invoked/src/test/java/org/kie/kogito/integrationtests/SpringBootTopicsInformationResourceTest.java
+++ b/springboot/addons/messaging/integration-tests/src/it/cloudevents-spring-boot-addon-it-invoked/src/test/java/org/kie/kogito/integrationtests/SpringBootTopicsInformationResourceTest.java
@@ -23,7 +23,7 @@ import org.kie.kogito.app.KogitoSpringbootApplication;
 import org.kie.kogito.event.ChannelType;
 import org.kie.kogito.event.Topic;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 
 import io.restassured.RestAssured;
 
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java
index 8f7aea6145..88fabb4e9b 100644
--- a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java
@@ -18,7 +18,7 @@ package org.kie.kogito.integrationtests.springboot;
 
 import io.restassured.RestAssured;
 import org.junit.jupiter.api.BeforeEach;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 
 public abstract class BaseRestTest {
 
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/ElementAtIndexTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/ElementAtIndexTest.java
index b7e764307a..77fd646743 100644
--- a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/ElementAtIndexTest.java
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/ElementAtIndexTest.java
@@ -20,7 +20,7 @@ import io.restassured.http.ContentType;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 import static io.restassured.RestAssured.given;
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-kafka-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-kafka-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java
index 8f7aea6145..88fabb4e9b 100644
--- a/springboot/integration-tests/src/it/integration-tests-springboot-kafka-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-kafka-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java
@@ -18,7 +18,7 @@ package org.kie.kogito.integrationtests.springboot;
 
 import io.restassured.RestAssured;
 import org.junit.jupiter.api.BeforeEach;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 
 public abstract class BaseRestTest {
 
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java
index 8f7aea6145..88fabb4e9b 100644
--- a/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/BaseRestTest.java
@@ -18,7 +18,7 @@ package org.kie.kogito.integrationtests.springboot;
 
 import io.restassured.RestAssured;
 import org.junit.jupiter.api.BeforeEach;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 
 public abstract class BaseRestTest {
 
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/ProcessTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/ProcessTest.java
index 3177fd3d5e..500b200ecd 100644
--- a/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/ProcessTest.java
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/ProcessTest.java
@@ -27,7 +27,7 @@ import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
 
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 import static io.restassured.RestAssured.given;
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java
index 83085a9be3..14a8f95d22 100644
--- a/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-processes-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java
@@ -37,7 +37,7 @@ import org.junit.jupiter.api.extension.ExtendWith;
 import org.kie.kogito.process.workitem.AttachmentInfo;
 import org.kie.kogito.task.management.service.TaskInfo;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 import static io.restassured.RestAssured.given;
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-processes-persistence-it/integration-tests-springboot-processes-infinispan/pom.xml b/springboot/integration-tests/src/it/integration-tests-springboot-processes-persistence-it/integration-tests-springboot-processes-infinispan/pom.xml
index c102fb4ad1..cda15d9596 100644
--- a/springboot/integration-tests/src/it/integration-tests-springboot-processes-persistence-it/integration-tests-springboot-processes-infinispan/pom.xml
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-processes-persistence-it/integration-tests-springboot-processes-infinispan/pom.xml
@@ -39,11 +39,10 @@
     </dependency>
     <dependency>
       <groupId>org.infinispan</groupId>
-      <artifactId>infinispan-spring-boot-starter-remote</artifactId>
+      <artifactId>infinispan-spring-boot3-starter-remote</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.kie.kogito</groupId>
-      <artifactId>kogito-addons-springboot-process-management</artifactId>
+      <groupId>org.kie.kogito</groupId>      <artifactId>kogito-addons-springboot-process-management</artifactId>
     </dependency>
     <dependency>
       <groupId>org.kie.kogito</groupId>
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-processes-persistence-it/integration-tests-springboot-processes-persistence-common/src/test/java/org/kie/kogito/it/PersistenceTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-processes-persistence-it/integration-tests-springboot-processes-persistence-common/src/test/java/org/kie/kogito/it/PersistenceTest.java
index 95775d9a7c..587345f7d8 100644
--- a/springboot/integration-tests/src/it/integration-tests-springboot-processes-persistence-it/integration-tests-springboot-processes-persistence-common/src/test/java/org/kie/kogito/it/PersistenceTest.java
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-processes-persistence-it/integration-tests-springboot-processes-persistence-common/src/test/java/org/kie/kogito/it/PersistenceTest.java
@@ -30,7 +30,7 @@ import org.kie.kogito.Address;
 import org.kie.kogito.AddressType;
 import org.kie.kogito.Person;
 import org.kie.kogito.Status;
-import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.boot.test.web.server.LocalServerPort;
 
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
diff --git a/springboot/starters/kogito-processes-spring-boot-starter/pom.xml b/springboot/starters/kogito-processes-spring-boot-starter/pom.xml
index 9eed010a53..627c1085c3 100644
--- a/springboot/starters/kogito-processes-spring-boot-starter/pom.xml
+++ b/springboot/starters/kogito-processes-spring-boot-starter/pom.xml
@@ -22,6 +22,10 @@
       <groupId>org.kie.kogito</groupId>
       <artifactId>kogito-rest-workitem</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.reactivestreams</groupId>
+      <artifactId>reactive-streams</artifactId>
+    </dependency>
     <dependency>
       <groupId>org.springframework.security</groupId>
       <artifactId>spring-security-core</artifactId>
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/PMMLTreeTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/PMMLTreeTest.java
--- a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/PMMLTreeTest.java	(revision 9e11adb52d8ca94fd2f25beebf07cd5925983ac4)
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/PMMLTreeTest.java	(date 1693288356561)
@@ -40,7 +40,7 @@
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = KogitoSpringbootApplication.class)
 class PMMLTreeTest extends BaseRestTest {

-    private static final String BASE_PATH = "/PMMLTree/SampleMine/";
+    private static final String BASE_PATH = "/PMMLTree/SampleMine";
     private static final String TARGET = "decision";

     static {
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/PMMLRegressionTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/PMMLRegressionTest.java
--- a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/PMMLRegressionTest.java	(revision 9e11adb52d8ca94fd2f25beebf07cd5925983ac4)
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/PMMLRegressionTest.java	(date 1693288356557)
@@ -43,7 +43,7 @@
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = KogitoSpringbootApplication.class)
 class PMMLRegressionTest extends BaseRestTest {

-    private static final String BASE_PATH = "/PMMLRegression/LinReg/";
+    private static final String BASE_PATH = "/PMMLRegression/LinReg";
     private static final String TARGET = "fld4";

     static {
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/CommonPMMLTestUtils.java b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/CommonPMMLTestUtils.java
--- a/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/CommonPMMLTestUtils.java	(revision 9e11adb52d8ca94fd2f25beebf07cd5925983ac4)
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-decisions-it/src/test/java/org/kie/kogito/integrationtests/springboot/CommonPMMLTestUtils.java	(date 1693288384011)
@@ -61,7 +61,7 @@
                                   final String basePath,
                                   final String targetField,
                                   final Map<String, Object> expectedResultMap) {
-        String path = basePath + "descriptive";
+        String path = basePath + "/descriptive";
         final Response response = given()
                 .contentType(ContentType.JSON)
                 .body(inputData)
diff --git a/springboot/integration-tests/src/it/integration-tests-springboot-kafka-it/src/test/java/org/kie/kogito/integrationtests/springboot/PingPongMessageTest.java b/springboot/integration-tests/src/it/integration-tests-springboot-kafka-it/src/test/java/org/kie/kogito/integrationtests/springboot/PingPongMessageTest.java
--- a/springboot/integration-tests/src/it/integration-tests-springboot-kafka-it/src/test/java/org/kie/kogito/integrationtests/springboot/PingPongMessageTest.java	(revision 9e11adb52d8ca94fd2f25beebf07cd5925983ac4)
+++ b/springboot/integration-tests/src/it/integration-tests-springboot-kafka-it/src/test/java/org/kie/kogito/integrationtests/springboot/PingPongMessageTest.java	(date 1693286486274)
@@ -82,7 +82,7 @@
                 .untilAsserted(() -> given()
                         .contentType(ContentType.JSON)
                         .when()
-                        .get("/pong_message/")
+                        .get("/pong_message")
                         .then()
                         .statusCode(200)
                         .body("$.size()", equalTo(1)));
@@ -90,7 +90,7 @@
         String pId = given()
                 .contentType(ContentType.JSON)
                 .when()
-                .get("/pong_message/")
+                .get("/pong_message")
                 .then()
                 .statusCode(200)
                 .body("$.size()", equalTo(1))
