commit f1d2242a0ab9e88a55d9e5b2b005dbe4d3c2f6b9
Author: Pere Fernández <pefernan@users.noreply.github.com>
Date:   Thu Jun 15 13:46:24 2023 +0200

    Fix Constructor signature

diff --git a/quarkus/addons/python/deployment/src/main/java/org/kie/kogito/quarkus/serverless/workflow/python/PythonAddonCapabilityProcessor.java b/quarkus/addons/python/deployment/src/main/java/org/kie/kogito/quarkus/serverless/workflow/python/PythonAddonCapabilityProcessor.java
index c267fed00d..0cd13580c3 100644
--- a/quarkus/addons/python/deployment/src/main/java/org/kie/kogito/quarkus/serverless/workflow/python/PythonAddonCapabilityProcessor.java
+++ b/quarkus/addons/python/deployment/src/main/java/org/kie/kogito/quarkus/serverless/workflow/python/PythonAddonCapabilityProcessor.java
@@ -40,6 +40,6 @@ class PythonAddonCapabilityProcessor extends OneOfCapabilityKogitoAddOnProcessor
 
     @BuildStep
     JniRuntimeAccessBuildItem jniRegister() {
-        return new JniRuntimeAccessBuildItem(true, true, true, true, SharedInterpreter.class);
+        return new JniRuntimeAccessBuildItem(true, true, true, SharedInterpreter.class);
     }
 }
